src/core/js2python.c

```diff
@@ -115,9 +116,24 @@ EM_JS_NUM(errcode, js2python_init, (), {
 
   Module.__js2python_bigint = function(value)
   {
-    let ptr = stringToNewUTF8(value.toString(16));
-    let result = _PyLong_FromString(ptr, 0, 16);
-    _free(ptr);
+    let value_orig = value;
+    let length = 0;
+    while (value && ~value) {
+      length++;
+      value >>= BigInt(32);
```

Comment:
Wouldn't it be faster to compute the length with `Math.log2(value)` instead of byte shifts?

Reply:
It doesn't seem to work:
```js
>>> Math.log2(1471057108571074017501n)
TypeError: can't convert BigInt to number
```

Reply:
It might be necessary to convert to double
```js
>>> Math.log2(Number(1471057108571074017501n))
```
but `log` is also expensive, so maybe the current version is better.